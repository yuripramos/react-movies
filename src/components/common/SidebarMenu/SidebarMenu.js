import React, { Component } from "react";
import {
  ListMenuItems,
  Item,
  Span,
  MenuList,
  Divisor,
  LeftBorder,
  ListItem
} from "./styles";
import { string, bool, func } from "prop-types";
import { redirect } from "../../../utils/redirect";

class SidebarMenu extends Component {
  constructor(props) {
    super(props);
    this.isActive = this.isActive.bind(this);
  }

  isActive(path) {
    const { currentLocation } = this.props;
    return currentLocation.includes(path);
  }

  redirectTo(path) {
    const { closeMenu } = this.props;
    closeMenu();
    redirect(path);
  }

  render() {
    const { desktop } = this.props;
    if (desktop) {
      return (
        <ListMenuItems>
          <Item
            onClick={() => redirect("/news")}
            active={this.isActive("/news")}
          >
            <Span>NEWS</Span>
          </Item>
          <Item
            onClick={() => redirect("/vision")}
            active={this.isActive("/vision")}
          >
            <Span>OUR VISION</Span>
          </Item>
          <Item
            onClick={() => redirect("/contact")}
            active={this.isActive("/contact")}
          >
            <Span>CONTACT</Span>
          </Item>
        </ListMenuItems>
      );
    } else {
      return (
        <MenuList>
          <Divisor />
          <ListItem
            onClick={() => redirect("/news")}
            active={this.isActive("/news")}
          >
            <LeftBorder />
            NEWS
          </ListItem>
          <ListItem
            onClick={() => redirect("/vision")}
            active={this.isActive("/vision")}
          >
            <LeftBorder />
            OUR VISION
          </ListItem>
          <ListItem
            onClick={() => redirect("/contact")}
            active={this.isActive("/contact")}
          >
            <LeftBorder />
            CONTACT
          </ListItem>
        </MenuList>
      );
    }
  }
}

export default SidebarMenu;

SidebarMenu.propTypes = {
  currentLocation: string,
  desktop: bool,
  closeMenu: func
};
