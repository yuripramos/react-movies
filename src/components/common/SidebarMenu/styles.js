import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";
import { white, trueWhite, darkestBlue, green } from "../../../styles/settings";
import Icon from "../Icon";

export const MenuList = styled.ul`
  background: ${white};
  height: 100vh;
`;

export const Divisor = styled.div`
  height: ${rem(25)};
  box-shadow: 0 ${rem(10)} ${rem(15)} ${rem(6)} rgba(159, 189, 204, 0.15);
  margin-bottom: ${rem(25)};
`;

export const ListItem = styled.li`
  cursor: pointer;
  position: relative;
  height: ${rem(50)};
  font-family: Roboto;
  font-size: ${rem(15)};
  color: ${darkestBlue};
  list-style: none;
  letter-spacing: ${rem(2)};
  padding-top: ${rem(5)};
  margin: 0 ${rem(5)} ${rem(5)} 0;
  text-align: left;

  ${({ active }) =>
    active &&
    css`
      color: ${green};
    `};

  :hover {
    opacity: 0.7;
  }
`;

export const LeftBorder = styled.div`
  width: ${rem(7)};
  height: 100%;
  background-color: ${darkestBlue};
  position: absolute;
  top: 0;
  left: -${rem(40)};
`;

export const Item = styled.div`
  cursor: pointer;
  flex-flow: row wrap;
  width: ${rem(80)};
  text-align: center;
  position: relative;
  border-top: solid ${rem(2)} transparent;

  ${media.md(css`
    width: ${rem(130)};
  `)};

  ${({ active }) =>
    !active &&
    css`
      :hover {
        color: ${white};
      }
    `};

  ${({ active }) =>
    active &&
    css`
      cursor: initial;
      color: ${white};
      font-weight: 600;
    `};
`;

export const MenuIcons = styled(Icon)`
  width: ${rem(18)};
  padding-top: ${rem(10)};
  padding-bottom: ${rem(9)};
  display: flex;
  align-items: center;
  justify-content: center;
  ${media.sm(css`
    width: auto;
  `)};
`;

export const Span = styled.span`
  margin-top: ${rem(9)};
  font-size: ${rem(12)};
`;

export const ListMenuItems = styled.div`
  margin: 0 auto;
  font-weight: 500;
  color: ${trueWhite};
  margin-top: ${rem(15)};
  display: flex;
  flex-flow: row wrap;
  height: ${rem(30)};
`;
