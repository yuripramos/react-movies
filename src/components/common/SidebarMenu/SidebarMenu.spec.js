import React from "react";
import { shallow } from "enzyme";

import SidebarMenu from "./SidebarMenu";

jest.mock("../../../utils/redirect");
const redirectMock = require("../../../utils/redirect").redirect;

const closeMenuMock = jest.fn();

const props = {
  currentLocation: "",
  closeMenu: closeMenuMock
};

describe("SidebarMenu component", () => {
  it("should match snapshot", () => {
    expect(shallow(<SidebarMenu {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot for desktop", () => {
    expect(shallow(<SidebarMenu {...props} desktop />)).toMatchSnapshot();
  });

  describe("redirect", () => {
    beforeEach(() => {
      const instance = shallow(<SidebarMenu {...props} />).instance();
      instance.redirectTo("/path");
    });

    it("should invoke closeMenu", () => {
      expect(closeMenuMock).toHaveBeenCalledWith();
    });

    it("should invoke redirect", () => {
      expect(redirectMock).toHaveBeenCalledWith("/path");
    });
  });

  describe("isActive", () => {
    it("should return true if currentLocation includes the path", () => {
      const instance = shallow(
        <SidebarMenu {...props} currentLocation={"/path"} />
      ).instance();
      expect(instance.isActive("path")).toBe(true);
    });

    it("should return false if currentLocation doesnt include the path", () => {
      const instance = shallow(<SidebarMenu {...props} />).instance();
      instance.isActive("lalalalala");
      expect(instance.isActive("path")).toBe(false);
    });
  });
});
