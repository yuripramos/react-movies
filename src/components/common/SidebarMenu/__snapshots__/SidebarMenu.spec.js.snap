// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarMenu component should match snapshot 1`] = `
<styled.ul>
  <styled.div />
  <styled.li
    active={false}
    onClick={[Function]}
  >
    <styled.div />
    NEWS
  </styled.li>
  <styled.li
    active={false}
    onClick={[Function]}
  >
    <styled.div />
    OUR VISION
  </styled.li>
  <styled.li
    active={false}
    onClick={[Function]}
  >
    <styled.div />
    CONTACT
  </styled.li>
</styled.ul>
`;

exports[`SidebarMenu component should match snapshot for desktop 1`] = `
<styled.div>
  <styled.div
    active={false}
    onClick={[Function]}
  >
    <styled.span>
      NEWS
    </styled.span>
  </styled.div>
  <styled.div
    active={false}
    onClick={[Function]}
  >
    <styled.span>
      OUR VISION
    </styled.span>
  </styled.div>
  <styled.div
    active={false}
    onClick={[Function]}
  >
    <styled.span>
      CONTACT
    </styled.span>
  </styled.div>
</styled.div>
`;
